### PR TITLE
blind commit of work, as I managed to break my local git repository

### PR DESCRIPTION
**Pull request by jsteemann**
Thanks git for the following:
```
error: object file .git/objects/b8/20b3d793c64f435d75644045ec45677a4029fc is empty
error: unable to mmap .git/objects/b8/20b3d793c64f435d75644045ec45677a4029fc: No such file or directory
error: b820b3d793c64f435d75644045ec45677a4029fc: object corrupt or missing: .git/objects/b8/20b3d793c64f435d75644045ec45677a4029fc
Checking object directories: 100% (256/256), done.
Checking objects: 100% (776213/776213), done.
error: refs/heads/bug-fix/remove-initialize-aql-api: invalid sha1 pointer 0000000000000000000000000000000000000000
error: bad ref for .git/logs/refs/heads/bug-fix/remove-initialize-aql-api
Checking connectivity: 773159, done.
dangling blob 46ec60d8743121a28d0de9f3d5abd5b2ba87fc19
dangling blob a01923e01fa5341951db4d46dc5f370be5136ccf
dangling tree d0c7838410ad4e069a377c6cd86accd33e7cd9c9
dangling blob 807025ef6202b8e84af3ff59b0aba35a6d6b28ec
dangling blob ca962929f333d369efaa372d6d85ddaab0040651
dangling blob 82baa97b5ca0b72fe3ab74651c0458e959d409da
dangling blob 2b8e0b4f6794077b8700d05451ff1db18f3c315d
dangling blob c8ef4bc5d3deb9837be90d2382b998a02ac3926c
dangling blob 36f08d4d9cac8c3221a31e7ae2c8ce770f155a27
dangling blob 86aece7d3006494232846255cee35aeaa6236c8c
dangling blob 03424f471174c84b4e2b6c28ae89169c8ff4552a
dangling blob aa5b5199b6751b878ff6e1f43e24a4b21506c5b0
dangling blob 00d7b1abbe61adefcbf82c6908d2c2822f8cbcef
missing blob b820b3d793c64f435d75644045ec45677a4029fc
dangling blob ca70147f114f0b232784e56f2df052169c864940
dangling blob f8cf14d0fd580070794f6d2034738feb4e754016
dangling blob d2eef6f24c11922b4c74411e2287ff9151f72dc8
dangling blob 3a63f8c2b6ad7052974d39d599147c261c936cbb
dangling blob 5ac7dcc14a064f08206e4639c0ec35381833530e
```

The PR is about removing the "initialize" call from AQL completely.
"initializeCursor" will now run lazily in the cluster, so that no explicit call to it is required for the simple cases.

Additionally, the unused "/_api/aql/skip" route was removed.

### CODE REVIEW

File: arangod/Aql/ClusterBlocks.cpp [hunk 1]
```diff
@@ -1209,71 +1200,37 @@ RemoteBlock::RemoteBlock(ExecutionEngine* engine, RemoteNode const* en,
...
-  throwExceptionAfterBadSyncRequest(res.get(), false);
-
-  // If we get here, then res->result is the response which will be
-  // a serialized AqlItemBlock:
-  StringBuffer const& responseBodyBuf(res->result->getBody());
-
-  std::shared_ptr<VPackBuilder> builder =
-      VPackParser::fromJson(responseBodyBuf.c_str(), responseBodyBuf.length());
-  VPackSlice slice = builder->slice();
+  ++_engine->_stats.httpRequests;
```
Comment (graetzer) [inline arangod/Aql/ClusterBlocks.cpp]: why don't we rename this to requests ?

File: arangod/Aql/ExecutionNode.cpp [hunk 2]
```diff
@@ -1569,6 +1569,45 @@ void CalculationNode::toVelocyPackHelper(VPackBuilder& nodes, unsigned flags) co
   }
 
   nodes.add("expressionType", VPackValue(_expression->typeString()));
+
```
Comment (graetzer) [inline arangod/Aql/ExecutionNode.cpp]: this is already in https://github.com/arangodb/arangodb/pull/5451/files

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/559/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/562/

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr-linux/568/

**Pull request merged**